demo/home/home-controller.js

```diff
@@ -23,13 +23,16 @@ app.controller('HomeCtrl', function($scope, Data, File, $timeout, $http, Caching
           if ($scope.vars[variable]) {
             return;
           }
-          var varFileName = $scope.questionsVariables[variable].queryFileName;
-          var file = 'data/query/' + varFileName;
-          var promise = Data.variables(file);
-          $scope.vars[variable] = promise;
-          promise.then(function(vars) {
-            $scope.vars[variable] = vars;
-          });
+          // Variable JSON config is not defined it is a free text input
+          if($scope.questionsVariables[variable] != undefined) {
```

Comment:
We could also use the [includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method of the Array prototype. Not every browser supports this, so that needs further investigation.